### PR TITLE
feat(enode): add show config api

### PR DESCRIPTION
**Pull request by hunshenshi**
issue #247

### CODE REVIEW

File: cmd/enode/api/http.go [hunk 1]
```diff
@@ -46,6 +51,19 @@ type queryMessageStateLogResp struct {
 	States    []*stateLog `json:"states"`
 }
 
+type configInfoResp struct {
+	ChainEndpoint          string `json:"chainEndpoint"`
+	HttpServerEndpoint     string `json:"httpServerEndpoint"`
+	DataBaseDSN            string `json:"dataBaseDSN"`
```
Comment (huangzhiran) [inline cmd/enode/api/http.go]: maybe DSN should not return, for it has password for pg

File: cmd/enode/api/http.go [hunk 2]
```diff
@@ -46,6 +51,19 @@ type queryMessageStateLogResp struct {
...
+	BootNodeMultiaddr      string `json:"bootNodeMultiaddr"`
+	IotexChainID           int    `json:"iotexChainID"`
+	ProjectContractAddress string `json:"projectContractAddress"`
+	IPFSEndpoint           string `json:"IPFSEndpoint"`
+	DIDAuthServerEndpoint  string `json:"DIDAuthServerEndpoint"`
+	OperatorIoTeXAddress   string `json:"OperatorIoTeXAddress"`
```
Comment (huangzhiran) [inline cmd/enode/api/http.go]: OperatorETHAddress

File: cmd/enode/api/http.go [hunk 3]
```diff
@@ -187,3 +206,33 @@ func (s *HttpServer) issueJWTCredential(c *gin.Context) {
 	c.JSON(http.StatusOK, rsp)
 	return
 }
+
+func (s *HttpServer) getConfigInfo(c *gin.Context) {
```
Comment (huangzhiran) [inline cmd/enode/api/http.go]: we can use viper.AllSettings() and don't need move env to a new packages

File: cmd/enode/api/http.go (hunk 1 above)
Comment (CoderZhi) [reply cmd/enode/api/http.go]: Database

File: cmd/enode/api/http.go [hunk 4]
```diff
@@ -46,6 +51,19 @@ type queryMessageStateLogResp struct {
 	States    []*stateLog `json:"states"`
 }
 
+type configInfoResp struct {
+	ChainEndpoint          string `json:"chainEndpoint"`
+	HttpServerEndpoint     string `json:"httpServerEndpoint"`
+	DataBaseDSN            string `json:"dataBaseDSN"`
+	BootNodeMultiaddr      string `json:"bootNodeMultiaddr"`
+	IotexChainID           int    `json:"iotexChainID"`
```
Comment (CoderZhi) [inline cmd/enode/api/http.go]: IoTeX

File: cmd/enode/api/http.go [hunk 5]
```diff
@@ -64,6 +82,7 @@ func NewHttpServer(pg *persistence.Postgres, didAuthServer string, projectManage
 	s.engine.POST("/message", s.handleMessage)
 	s.engine.GET("/message/:id", s.queryStateLogByID)
 	s.engine.POST("/sign_credential", s.issueJWTCredential)
+	s.engine.GET("/config", s.getConfigInfo)
```
Comment (CoderZhi) [inline cmd/enode/api/http.go]: `s.engine.GET("/project_config/:projectID", s.getProjectConfig)`

File: cmd/enode/api/http.go (hunk 3 above)
Comment (CoderZhi) [reply cmd/enode/api/http.go]: getProjectConfig(c *gin.Context)

File: cmd/enode/api/http.go [hunk 6]
```diff
@@ -46,6 +51,19 @@ type queryMessageStateLogResp struct {
 	States    []*stateLog `json:"states"`
 }
 
+type configInfoResp struct {
```
Comment (CoderZhi) [inline cmd/enode/api/http.go]: this is more like an admin endpoint response if some sensitive settings are returned.

File: cmd/enode/api/http.go [hunk 7]
```diff
@@ -8,13 +8,12 @@ import (
...
 	"github.com/gin-gonic/gin"
 	"github.com/google/uuid"
-	"github.com/pkg/errors"
-
 	"github.com/machinefi/sprout/auth/didvc"
 	"github.com/machinefi/sprout/clients"
 	"github.com/machinefi/sprout/persistence"
 	"github.com/machinefi/sprout/project"
 	"github.com/machinefi/sprout/types"
+	"github.com/pkg/errors"
```
Comment (huangzhiran) [inline cmd/enode/api/http.go:16]: move this to second block, for it is outer depends

File: cmd/enode/api/http.go [hunk 8]
```diff
@@ -46,24 +45,34 @@ type queryMessageStateLogResp struct {
 	States    []*stateLog `json:"states"`
 }
 
+type ENodeConfigInfoResp struct {
```
Comment (huangzhiran) [inline cmd/enode/api/http.go:48]: ENodeConfigResp

File: cmd/enode/api/http.go [hunk 9]
```diff
@@ -46,24 +45,34 @@ type queryMessageStateLogResp struct {
...
+	ProjectContractAddress string `json:"projectContractAddress"`
+	OperatorETHAddress     string `json:"OperatorETHAddress"`
```
Comment (huangzhiran) [inline cmd/enode/api/http.go:50]: `json:"OperatorETHAddress,omitempty"` for it's may empty. same as below

**Review by huangzhiran — APPROVED** [review]

**Pull request merged**